cryptarchia/cryptarchia.py

```diff
@@ -46,8 +46,30 @@ class Config:
 class BlockHeader:
     slot: Slot
     parent: Id
-    # TODO: spec out the block id, this is just a placeholder to unblock tests
-    id: Id
+    content_size: int
+    content_id: Id
+
+    # **Attention**:
+    # The ID of a block header is defined as the 32byte blake2b hash of its fields
+    # as serialized in the format specified by the 'HEADER' rule in 'messages.abnf'.
+    #
+    # The following code is to be considered as a reference implementation, mostly to be used for testing.
+    def id(self) -> Id:
+        # version byte
+        bytes = bytearray(b"\x01")
+        # header type
+        bytes += b"\x00"
+        # content size
+        bytes += int.to_bytes(self.content_size, length=4, byteorder="big")
+        # content id
+        assert len(self.content_id) == 32
+        bytes += self.content_id
+        # slot
+        bytes += int.to_bytes(self.slot.absolute_slot, length=8, byteorder="big")
+        # parent
+        assert len(self.content_id) == 32
```

Comment:
I think we don’t need this assertion here again because we have it in line 65

Reply:
yes, good catch